tests/simulator/test_initialization.py

```diff
@@ -582,5 +585,58 @@ def test_overlapped_particledatas_have_identical_particles(self, interp_order, r
 
 
 
+    def _test_levelghostparticles_have_correct_split_from_coarser_particle(self, dim, interp_order, refinement_boxes):
+        print("test_levelghostparticles_have_correct_split_from_coarser_particle for dim/interp : {}/{}".format(dim, interp_order))
+
+        datahier = self.getHierarchy(interp_order, refinement_boxes, "particles")
+        from pyphare.pharein.global_vars import sim
+        assert sim is not None
+
+        def domainParticles_for(ilvl, particles_id):
+            patchDatas = [
+              patch.patch_datas[particles_id]
+                  for patch in datahier.levels()[ilvl].patches
+                      if particles_id in patch.patch_datas
+            ]
+            return aggregate_particles([
+              patchData.dataset.select(patchData.box) for patchData in patchDatas
+            ]) # including patch ghost particles means duplicates
+
+        for ilvl, particle_gaboxes in level_ghost_boxes(datahier).items():
+            assert ilvl > 0 # has no level 0
+
+            for particles_id, gaboxes_list in particle_gaboxes.items():
+                coarse_particles = domainParticles_for(ilvl - 1, particles_id)
+                coarse_split_particles = coarse_particles.split(sim)
+                assert coarse_split_particles.size() == coarse_particles.size() * sim.refined_particle_nbr
```

Comment:
I think we need to assert sizes are not zero also?

Reply:
ok!